beets/dbcore/query.py

```diff
@@ -146,10 +146,9 @@ def __init__(self, field, fast=True):
     def col_clause(self):
         return self.field + " IS NULL", ()
 
-    @classmethod
     def match(self, item):
         try:
-            return item[self.field] is None
+            return item[self.field] is None or not item[self.field]
```

Comment:
"IS NULL" works for fixed fields, which always exist, but not for flexible fields (which use `match` instead anyway).